### PR TITLE
Add template module.

### PR DESCRIPTION
**Pull request by maltoe**
Screenshot 2020-03-13 at 00 12 53

### CODE REVIEW

File: lib/chromic_pdf/pdf/template.ex [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      header_height: Keyword.get(opts, :header_height, "0"),
+      header_font_size: Keyword.get(opts, :header_font_size, "10pt"),
+      footer_height: Keyword.get(opts, :footer_height, "0"),
+      footer_font_size: Keyword.get(opts, :footer_font_size, "10pt")
+    ]
+
+    do_render_styles(assigns)
+  end
+
+  EEx.function_from_string(:defp, :do_render_styles, @styles, [:assigns])
```
Comment (nickgnd) [inline lib/chromic_pdf/pdf/template.ex]: 💡

File: lib/chromic_pdf/pdf/template.ex [hunk 2]
```diff
@@ -0,0 +1,172 @@
+defmodule ChromicPDF.Template do
+  @moduledoc """
+  This module contains basic styles and options that make it simple to make the HTML content fully
+  cover a given page.
+  """
```
Comment (nickgnd) [inline lib/chromic_pdf/pdf/template.ex]: This does not sound fully clear to me.
I would probably mention that this module allows you to create a reusable template with custom content such header, footer, style and even content.

Something like:

"Use this module to create your HTML template with a custom header, footer, content and style.

or

"This module exposes helper methods to create your own HTML page template, including header, footer, content and style."


Feel free to ignore my comment if it is incorrect 😉

File: lib/chromic_pdf/pdf/template.ex [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+  require EEx
+
+  @type blob :: binary()
+
+  @type content_option ::
+          {:html, blob()}
```
Comment (nickgnd) [inline lib/chromic_pdf/pdf/template.ex]: Why did you choose `:html` for this variable name?
I fear that it can lead to confusion. `:header` and `:footer` are valid HTML too.
What about `:body`, `:page_content`, or simply `:content`?

File: lib/chromic_pdf/pdf/template.ex [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  <h2>User Agent</h2>
+  <pre id="user-agent"></pre>
+
+  <script type="text/javascript">
+  window.onload = function() {
+    var browser, userAgent = navigator.userAgent;
+    document.getElementById('user-agent').innerHTML = userAgent;
+  };
+  </script>
+  """
```
Comment (nickgnd) [inline lib/chromic_pdf/pdf/template.ex]: Cool!

File: lib/chromic_pdf/pdf/template.ex [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+      height: Keyword.get(opts, :height, "215.9mm"),
+      width: Keyword.get(opts, :width, "279.4mm"),
+      header_height: Keyword.get(opts, :header_height, "0"),
+      header_font_size: Keyword.get(opts, :header_font_size, "10pt"),
+      footer_height: Keyword.get(opts, :footer_height, "0"),
+      footer_font_size: Keyword.get(opts, :footer_font_size, "10pt")
+    ]
+
+    do_render_styles(assigns)
+  end
```
Comment (nickgnd) [inline lib/chromic_pdf/pdf/template.ex]: Mmhh 🤔 
This seems pretty strict to me. Basically you can customize only height and font size of both header and footer, do I understand well? What if you have a more convoluted footer with different texts?
Do you need to style it inline?

File: lib/chromic_pdf/pdf/template.ex (hunk 3 above)
Comment (maltoe) [reply lib/chromic_pdf/pdf/template.ex]: 🤔 I wanted to align it with [`ChromicPDF.print_to_pdf/2`](https://github.com/bitcrowd/chromic_pdf/blob/0f0064c5c84c3ff6b055c873ffd885f2a3ebfc5e/lib/chromic_pdf/supervisor.ex#L114) where I also called it `:html`.

But you're right, it is indeed confusing and might even be a bit misleading in this case, as the param should only be body content markup, and don't have the enclosing `<html></html>` tags... Chromium has this print template in which they call it `content`, so I might go with this (and change the `:html` in `print_to_pdf/2`, as well) https://cs.chromium.org/chromium/src/components/printing/resources/print_header_footer_template_page.html?g=0&l=125

File: lib/chromic_pdf/pdf/template.ex (hunk 2 above)
Comment (maltoe) [reply lib/chromic_pdf/pdf/template.ex]: :) hmm, "create your own HTML page template" -> I would guess there is more to it than this file, so I'd like to keep it closer to "these are some helpers that help configure the page layout and margins" or so. Nevertheless, it's unclear, let me rephrase it 👍

File: lib/chromic_pdf/pdf/template.ex (hunk 5 above)
Comment (maltoe) [reply lib/chromic_pdf/pdf/template.ex]: Yes, the idea here, or the current functionality of `Template`, is only to give you configuration + absolutely minimum styles to easily create a page that has visible & consistent header & footer, and no weird other paddings/margins anywhere. It's surprisingly difficult to do, at first :)

All client styles and more layout styles need to be passed to `:html`, `:header`, `:footer` as `<style>` tags. I'll work on the documentation a bit.

thanks for your review 💚

File: lib/chromic_pdf/pdf/template.ex (hunk 3 above)
Comment (maltoe) [reply lib/chromic_pdf/pdf/template.ex]: hmm realized the param is actually called `html` in devtools, so going to keep it htere. But can still change it in the template, because why not 🤷‍♂️

https://chromedevtools.github.io/devtools-protocol/tot/Page#method-setDocumentContent

**Comment by maltoe** [comment]
@nickgnd updated the docs a bit, better like this?
